src/odemis/acq/test/flim_test.py

```diff
@@ -0,0 +1,205 @@
+
+Copyright © 2018 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+import logging
+import odemis
+from odemis import model
+import time
+import os
+import unittest
+from unittest.case import skip
+from odemis.util import test
+from odemis.acq import stream
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+SECOM_FLIM_CONFIG = CONFIG_PATH + "sim/secom-sim-flim.odm.yaml"
+
+'''
+To run these tests, you must run the Symphotime simulator.
+
+The simulator can be found at https://github.com/PicoQuant/SymPhoTime_Remote_Interface
+The PQ_ServerSimulator application is a Win32 or x64 based exe that simulates a Symphotime server
+By default, it listens on 127.0.0.1:6000 for a connection. Just launch the EXE with default settings
+in order to run these tests.
+'''
+
+
+class TestFlimtatic(unittest.TestCase):
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+
+        try:
+            test.start_backend(SECOM_FLIM_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Find CCD & SEM components
+        cls.sft = model.getComponent(role="time-correlator")
+        cls.tc_scanner = model.getComponent(role="tc-scanner")
+        cls.ex_light = model.getComponent(role="light")
+        cls.lscanner = model.getComponent(role="laser-mirror")
+        cls.apd = model.getComponent(role="tc-detector")
+        cls.detector = model.getComponent(role="photo-detector0")
+
+        cls.ex_light.period.value = 0.0001
+        em = [0] * len(cls.ex_light.emissions.value)
+        cls.ex_light.emissions.value = em
+        cls.ex_light.power.value = 0.25
+
+    @classmethod
+    def tearDownClass(cls):
+        time.sleep(1.0)
+
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def _on_image(self, im):
+        # Is called when an image is received in the live setting stream
+        logging.info("image: dim %s" , im.shape)
+        self._image = im
+
+    def validate_scan_sim(self, remote, repetition):
+        self.assertIsNotNone(remote.raw)
```

Comment:
You can check so many more things here!
At least, check that the raw[0].shape is the same as repetition[::-1].
You can also check that some of most important metadata is present and correct. eg, the `MD_DWELL_TIME` should be the `helper.dwellTime.value`. (BTW, that's probably not the case yet, as it's not updated when you integrate the data). You should check that `MD_PIXEL_SIZE` is (almost) the same as `helper.pixelSize.value`. You could even check that `MD_POS` is correct wrt the `.roi` (but that's quite a bit more involved).

Reply:
I made modifications to the checks, but somehow the values for pixel size and repetitions are not as expected. I will have to ask you about how this should work.

Reply:
It's good that there is a check then!
Get the pixelSize and dwell time as arguments too. BTW, you could also check MD_POS.
As you compute the accumulation of DataArrays of the right shape and (*in theory*) the right metadata, I'd expect that MD_POS, MD_PIXEL_SIZE and the shape should be fine out of the box. MD_DWELL_TIME needs to be updated explicitly though.